pareto/tests/test_data.py

```diff
@@ -0,0 +1,185 @@
+    ]
+    strategic = [
+        "Units",
+        "PNA",
+        "CNA",
+        "CCA",
+        "NNA",
+        "NCA",
+        "NKA",
+        "NRA",
+        "NSA",
+        "FCA",
+        "RCA",
+        "RNA",
+        "RSA",
+        "SCA",
+        "SNA",
+        "ROA",
+        "SOA",
+        "NOA",
+        "PCT",
+        "PKT",
+        "FCT",
+        "CST",
+        "CCT",
+        "CKT",
+        "RST",
+        "ROT",
+        "SOT",
+        "Elevation",
+        "CompletionsPadOutsideSystem",
+        "DesalinationTechnologies",
+        "DesalinationSites",
+        "BeneficialReuseCredit",
+        "TruckingTime",
+        "CompletionsDemand",
+        "PadRates",
+        "FlowbackRates",
+        "WellPressure",
+        "NodeCapacities",
+        "InitialPipelineCapacity",
+        "InitialPipelineDiameters",
+        "InitialDisposalCapacity",
+        "InitialTreatmentCapacity",
+        "ReuseMinimum",
+        "ReuseCapacity",
+        "FreshwaterSourcingAvailability",
+        "PadOffloadingCapacity",
+        "CompletionsPadStorage",
+        "DisposalOperationalCost",
+        "TreatmentOperationalCost",
+        "ReuseOperationalCost",
+        "PipelineOperationalCost",
+        "FreshSourcingCost",
+        "TruckingHourlyCost",
+        "PipelineDiameterValues",
+        "DisposalCapacityIncrements",
+        "InitialStorageCapacity",
+        "StorageCapacityIncrements",
+        "TreatmentCapacityIncrements",
+        "TreatmentEfficiency",
+        "RemovalEfficiency",
+        "DisposalExpansionCost",
+        "StorageExpansionCost",
+        "TreatmentExpansionCost",
+        "PipelineCapexDistanceBased",
+        "PipelineCapexCapacityBased",
+        "PipelineCapacityIncrements",
+        "PipelineExpansionDistance",
+        "Hydraulics",
+        "Economics",
+        "PadWaterQuality",
+        "StorageInitialWaterQuality",
+        "PadStorageInitialWaterQuality",
+        "DisposalOperatingCapacity",
+        "TreatmentExpansionLeadTime",
+        "DisposalExpansionLeadTime",
+        "StorageExpansionLeadTime",
+        "PipelineExpansionLeadTime_Dist",
+        "PipelineExpansionLeadTime_Capac",
+    ]
+
+
+@pytest.mark.parametrize(
+    "file_name",
+    [
+        "operational_generic_case_study.xlsx",
+        "strategic_toy_case_study.xlsx",
+        "strategic_treatment_demo.xlsx",
+        "strategic_small_case_study.xlsx",
+        "strategic_permian_demo.xlsx",
+    ],
+)
+def test_case_studies(file_name: str):
+    if file_name.startswith("operational"):
+        sets, parameters = Sets.operational, Parameters.operational
+    elif file_name.startswith("strategic"):
+        sets, parameters = Sets.strategic, Parameters.strategic
+
+    with resources.path("pareto.case_studies", file_name) as fpath:
+        # changing to raises=True will cause the test to fail on data loading failures
+        data = get_data(fpath, sets, parameters, raises=False)
```

Comment:
I agree that, if we were expecting no failures, `raises=True` would be the best choice. However, at the moment, `raises=True` would cause these tests to fail:

```
pareto/tests/test_data.py::test_case_studies[operational_generic_case_study.xlsx]
  /home/ludo/lbl/pareto/project-pareto/pareto/utilities/get_data.py:88: UserWarning: Data loading failed for the following sheets: 
  	PRT                   	Passed header=[1], len of 1, but only 1 lines in file (sheet: PRT)
  	CRT                   	Passed header=[1], len of 1, but only 1 lines in file (sheet: CRT)
  	TreatmentCapacity     	Passed header=[1], len of 1, but only 1 lines in file (sheet: TreatmentCapacity)
  	ProductionTankCapacity	Passed header=[1], len of 1, but only 1 lines in file (sheet: ProductionTankCapacity)
    warnings.warn(exc.summary)

pareto/tests/test_data.py: 106 warnings
  /home/ludo/lbl/pareto/project-pareto/pareto/utilities/get_data.py:228: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
    _df_parameters[i].replace("", np.nan, inplace=True)

pareto/tests/test_data.py::test_case_studies[strategic_toy_case_study.xlsx]
pareto/tests/test_data.py::test_case_studies[strategic_treatment_demo.xlsx]
pareto/tests/test_data.py::test_case_studies[strategic_small_case_study.xlsx]
pareto/tests/test_data.py::test_case_studies[strategic_permian_demo.xlsx]
  /home/ludo/lbl/pareto/project-pareto/pareto/tests/test_data.py:199: UserWarning: ProductionTanks is not found in defined sets or parameters but is parsed in the input data
    data = get_data(fpath, sets, parameters, raises=False)

pareto/tests/test_data.py::test_case_studies[strategic_toy_case_study.xlsx]
  /home/ludo/lbl/pareto/project-pareto/pareto/utilities/get_data.py:88: UserWarning: Data loading failed for the following sheets: 
  	RKA	Passed header=[1], len of 1, but only 1 lines in file (sheet: RKA)
  	RKT	Passed header=[1], len of 1, but only 1 lines in file (sheet: RKT)
    warnings.warn(exc.summary)

pareto/tests/test_data.py::test_case_studies[strategic_treatment_demo.xlsx]
pareto/tests/test_data.py::test_case_studies[strategic_small_case_study.xlsx]
pareto/tests/test_data.py::test_case_studies[strategic_permian_demo.xlsx]
  /home/ludo/lbl/pareto/project-pareto/pareto/utilities/get_data.py:88: UserWarning: Data loading failed for the following sheets: 
  	NOA                  	Passed header=[1], len of 1, but only 1 lines in file (sheet: NOA)
  	SOA                  	Passed header=[1], len of 1, but only 1 lines in file (sheet: SOA)
  	ROA                  	Passed header=[1], len of 1, but only 1 lines in file (sheet: ROA)
  	RKA                  	Passed header=[1], len of 1, but only 1 lines in file (sheet: RKA)
  	ROT                  	Passed header=[1], len of 1, but only 1 lines in file (sheet: ROT)
  	SOT                  	Passed header=[1], len of 1, but only 1 lines in file (sheet: SOT)
  	RKT                  	Passed header=[1], len of 1, but only 1 lines in file (sheet: RKT)
  	ReuseMinimum         	Passed header=[1], len of 1, but only 1 lines in file (sheet: ReuseMinimum)
  	ReuseCapacity        	Passed header=[1], len of 1, but only 1 lines in file (sheet: ReuseCapacity)
  	BeneficialReuseCost  	Passed header=[1], len of 1, but only 1 lines in file (sheet: BeneficialReuseCost)
  	BeneficialReuseCredit	Passed header=[1], len of 1, but only 1 lines in file (sheet: BeneficialReuseCredit)
  	SWDDeep              	Passed header=[1], len of 1, but only 1 lines in file (sheet: SWDDeep)
  	SWDAveragePressure   	Passed header=[1], len of 1, but only 1 lines in file (sheet: SWDAveragePressure)
  	SWDProxPAWell        	Passed header=[1], len of 1, but only 1 lines in file (sheet: SWDProxPAWell)
  	SWDProxInactiveWell  	Passed header=[1], len of 1, but only 1 lines in file (sheet: SWDProxInactiveWell)
  	SWDProxEQ            	Passed header=[1], len of 1, but only 1 lines in file (sheet: SWDProxEQ)
  	SWDProxFault         	Passed header=[1], len of 1, but only 1 lines in file (sheet: SWDProxFault)
  	SWDProxHpOrLpWell    	Passed header=[1], len of 1, but only 1 lines in file (sheet: SWDProxHpOrLpWell)
    warnings.warn(exc.summary)
```